client/src/app/videos/+video-edit/video-upload-guard.ts

```diff
@@ -0,0 +1,15 @@
+import { Injectable } from '@angular/core'
+import { CanDeactivate, ActivatedRouteSnapshot, RouterStateSnapshot } from '@angular/router'
+
+import { VideoAddComponent } from './video-add.component'
+
+@Injectable()
+export class VideoUploadGuard implements CanDeactivate<VideoAddComponent> {
+  canDeactivate (component: VideoAddComponent,
+    currentRoute: ActivatedRouteSnapshot,
+    currentState: RouterStateSnapshot,
+    nextState: RouterStateSnapshot
+  ): boolean {
+    return component.canDeactivate() || window.confirm('Your upload will be canceled, are you sure?')
```

Comment:
Maybe you could use the `confirmService`? :)

Usage example: https://github.com/Chocobozzz/PeerTube/blob/develop/client/src/app/videos/%2Bvideo-watch/video-watch.component.ts#L127